spec/decorators/article_decorator_spec.rb

```diff
@@ -183,4 +183,39 @@ def create_article(*args)
         description_and_tags).to eq(search_optimized_description_replacement)
     end
   end
+
+  describe "#video_metadata" do
+    let(:article_with_video) { create(:article) }
+    let(:video_data) do
+      {
+        id: article_with_video.id,
+        video_code: "ABC",
+        video_source_url: "https://cdn.com/ABC.m3u8",
+        video_thumbnail_url: "https://cdn.com/ABC.png",
+        video_closed_caption_track_url: "https://cdn.com/ABC_captions"
+      }
+    end
+
+    it "responds with a hash representation of video metadata" do
+      caption_track_url = video_data[:video_closed_caption_track_url]
+
+      allow(article_with_video).to receive(:video_code).
+        and_return(video_data[:video_code])
+      allow(article_with_video).to receive(:video_source_url).
+        and_return(video_data[:video_source_url])
+      allow(article_with_video).to receive(:cloudinary_video_url).
+        and_return(video_data[:video_thumbnail_url])
+      allow(article_with_video).to receive(:video_closed_caption_track_url).
+        and_return(caption_track_url)
```

Comment:
I don't think stubbing method ActiveRecord functions create a dependable test. You can simplify this whole process by generated articles with video links.

```
article_with_video = create(:article, video_code: "ABC", video_source_url: "https://cdn.com/ABC.m3u8", video_thumbnail_url: "https://cdn.com/ABC.png", video_closed_caption_track_url: "https://cdn.com/ABC_captions")
expect(article_with_video.decorate.video_metadata).to eq({
  id: article_with_video.id,
  ...
})
```

Reply:
This is so true! Just updated the spec to implement this